core/src/test/scala/integration/kafka/api/ConsumerTopicCreationTest.scala

```diff
@@ -35,9 +35,21 @@ import java.util.Collections
  * Tests behavior of specifying auto topic creation configuration for the consumer and broker
  */
 class ConsumerTopicCreationTest {
-  @ParameterizedTest
-  @MethodSource(Array("parameters"))
-  def testAutoTopicCreation(brokerAutoTopicCreationEnable: JBoolean, consumerAllowAutoCreateTopics: JBoolean): Unit = {
+
+
+  @Test
+  def testAutoCreationEnabled(): Unit = testAutoTopicCreation(true, true)
+
+  @Test
+  def testBrokerAutoCreationEnabled(): Unit = testAutoTopicCreation(true, false)
+
+  @Test
+  def testClientAutoCreationEnabled(): Unit = testAutoTopicCreation(false, true)
+
+  @Test
+  def testAutoCreationDisabled(): Unit = testAutoTopicCreation(false, false)
+
+  private def testAutoTopicCreation(brokerAutoTopicCreationEnable: JBoolean, consumerAllowAutoCreateTopics: JBoolean): Unit = {
```

Comment:
With this, you don't need the separate TestCase, right? You can rely on standard JUnit lifecycle methods.

Reply:
Oh, this doesn't solve the problem. In that case, my suggestion was bad and what you had before was better. Sorry about that.

Reply:
the related code is reverted.